src/services/EthereumService.ts

```diff
@@ -349,6 +349,7 @@ export class EthereumService {
 
   private async addWalletProviderListeners(): Promise<void> {
     await this.ensureMetaMaskWalletProvider();
+    await this.removeWalletProviderListeners();
```

Comment:
it seems the `this.removeWalletProviderListeners();` method already has `this.ensureMetaMaskWalletProvider()` inside it. Should we just remove the one above?

Reply:
I thought so too, but I thought, that both the `add...` and the `remove...` methods could also be called separately, so I just wanted to be overly protective.
(I was kind of experimenting with this `ensureXYZ` pattern, and I like it :D )

Maybe changing the order of the methods would make more sense?

Reply:
yep. changing order I think will make more sense